src/components/message/index.js

```diff
@@ -0,0 +1,98 @@
+import Message from "./Message.svelte";
+import { get } from "svelte/store";
+import messageStore from "./store";
+import { nanoid } from "nanoid";
+
+if (
+  typeof document !== "undefined" &&
+  !document.getElementById("svant-messages-wrapper")
+) {
+  // Create the component if it doesn't exist
+  new Message({ target: document.body });
+
+  const createMessageFunction = type => options => {
```

Comment:
This function does seem really complicated and it is hard to follow.

Maybe one thing we could do is think of how we could split this up a bit, and also for the top level function (not the inner destroy function) we could maybe create a variable that stores the current `messageStore` so you don't have to always do `get(messageStore)`.

Reply:
Yup makes sense. It is a big function 😅. I'll break it up.

Reply:
I'll add some comments too.

Reply:
I ended up only adding comments. I think breaking it up would make it more complex. Also extracting `get(messageStore)` to a function was reading funny, so I left it as is.

https://github.com/svelte-components/svant/pull/21/commits/9e4e548110178312711a82a70e286f511ec0d12c